lib/spack/spack/build_systems/python.py

```diff
@@ -359,30 +365,11 @@ def check_args(self, spec, prefix):
 
     # Testing
 
-    def build_test(self):
-        """Run unit tests after in-place build.
-
-        These tests are only run if the package actually has a 'test' command.
-        """
-        if self._setup_command_available('test'):
-            args = self.test_args(self.spec, self.prefix)
-
-            self.setup_py('test', *args)
-
-    def test_args(self, spec, prefix):
-        """Arguments to pass to test."""
-        return []
-
-    run_after('build')(PackageBase._run_default_build_time_test_callbacks)
-
-    def import_module_test(self):
-        """Attempts to import the module that was just installed.
-
-        This test is only run if the package overrides
-        :py:attr:`import_modules` with a list of module names."""
+    def test(self):
+        """Attempts to import modules of the installed package."""
 
         # Make sure we are importing the installed modules,
-        # not the ones in the current directory
+        # not the ones in the source directory
```

Comment:
Why not leverage the smoke test's `run_test` method to leverage the infrastructure @becker33 and I put in place for tracking and supporting multi-part tests?

Reply:
Done.